stacks-signer/src/signer.rs

```diff
@@ -632,14 +636,9 @@ impl Signer {
         block: &NakamotoBlock,
         current_reward_cycle: u64,
     ) -> bool {
-        let aggregate_key = retry_with_exponential_backoff(|| {
-            stacks_client
-                .get_approved_aggregate_key(self.reward_cycle)
-                .map_err(backoff::Error::transient)
-        })
-        .unwrap_or(None);
-        if aggregate_key.is_some() {
+        if self.approved_aggregate_public_key.is_some() {
             // We do not enforce a block contain any transactions except the aggregate votes when it is NOT already set
+            // TODO: should be only allow special cased transactions during prepare phase before a key is set?
```

Comment:
No; we just need to prioritize them over all others. If there's leftover block space, then other txs should be used to fill it.